src/System.Data.SqlClient/src/System/Data/SqlClient/TdsParser.Windows.cs

```diff
@@ -0,0 +1,124 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Data.SqlClient.SNI;
+
+namespace System.Data.SqlClient
+{
+    sealed internal partial class TdsParser
+    {
+        private static bool s_fSSPILoaded = false; // bool to indicate whether library has been loaded
+
+        internal void PostReadAsyncForMars()
+        {
+            if (TdsParserStateObjectFactory.useManagedSni)
+                return;
+
+            // HACK HACK HACK - for Async only
+            // Have to post read to initialize MARS - will get callback on this when connection goes
+            // down or is closed.
+
+            IntPtr temp = IntPtr.Zero;
+            uint error = TdsEnums.SNI_SUCCESS;
+
+            try { }
+            finally
```

Comment:
Nit: there aren't thread aborts in coreclr.  You could get rid of the try { } finally.